### DISCUSSION TITLE
Can't get openvpn-credentials to work

### DISCUSSION THREAD

**Discussion by locutus2000** [haugene/docker-transmission-openvpn#2078] [Q&A]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [ ] I have tried using dev branch

### Config used

  transmission-openvpn:
    cap_add:      
      - NET_ADMIN
    volumes:
      - d:/:/data
      - d:/Docker/vpn/:/etc/openvpn/custom/
      - d:/Docker/vpn/openvpn-credentials.txt:/config/openvpn-credentials.txt
    environment:
      - OPENVPN_PROVIDER=CUSTOM
      - OPENVPN_USERNAME=**None**
      - OPENVPN_PASSWORD=**None**
      - LOCAL_NETWORK=192.168.0.0/24
      - TRANSMISSION_DOWNLOAD_DIR=/data/Downloads/complete
      - TRANSMISSION_DOWNLOAD_QUEUE_SIZE=500
      - TRANSMISSION_IDLE_SEEDING_LIMIT=1
      - TRANSMISSION_RATIO_LIMIT=0
      - TRANSMISSION_WATCH_DIR=/data/NZB
      - TRANSMISSION_WATCH_DIR_ENABLED=true
      - PUID=1000
      - PGID=1000
    logging:
      driver: json-file
      options:
        max-size: 10m
    ports:
       - '9091:9091'
    image: haugene/transmission-openvpn

### Current Behavior

I can see that the container rewriters the default.ovpn file to point it to the user/pass file but I still get this error. But even when I just put in the username and password I get the same error.

### Expected Behavior

I was expecting a succesful connection.

### How have you tried to solve the problem?

I tried to add user and pass to the docker-compose file. I changed the password to contain only alfanumerics.

### Log output

Starting container with revision: 25b9724178f48227084f5a462b82b1fbc087498d

Creating TUN device /dev/net/tun

Using OpenVPN provider: CUSTOM

Running with VPN_CONFIG_SOURCE auto

No bundled config script found for CUSTOM. Defaulting to external config

Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.OJrY1QayHN

Extracting configs to /tmp/tmp.b8LGh2fLOt

ERROR: Could not find any configs for provider CUSTOM in downloaded configs

Cleanup: deleting /tmp/tmp.OJrY1QayHN and /tmp/tmp.b8LGh2fLOt

No VPN configuration provided. Using default.

Modifying /etc/openvpn/custom/default.ovpn for best behaviour in this container

Modification: Point auth-user-pass option to the username/password file

Modification: Change ca certificate path

Modification: Change ping options

Modification: Update/set resolv-retry to 15 seconds

Modification: Change tls-crypt keyfile path

Modification: Set output verbosity to 3

Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop

Found existing OPENVPN credentials at /config/openvpn-credentials.txt

adding route to local network 192.168.0.0/24 via 172.21.0.1 dev eth0

Thu Nov 25 08:42:50 2021 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 19 2021

Thu Nov 25 08:42:50 2021 library versions: OpenSSL 1.1.1f 31 Mar 2020, LZO 2.10

Thu Nov 25 08:42:50 2021 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts

Thu Nov 25 08:42:50 2021 neither stdin nor stderr are a tty device and you have neither a controlling tty nor systemd - can't ask for 'Enter Private Key Password:'. If you used --daemon, you need to use --askpass to make passphrase-protected keys work, and you can not use --auth-nocache.

Thu Nov 25 08:42:50 2021 Exiting due to fatal error

### Environment

```markdown
- OS:Windows 10 with WSL2
- Docker:v20.10.7
```

**Comment by pkishino** [comment]
Not a container issue..
The problem is with your ovpn config, check the log:
```
Thu Nov 25 08:42:50 2021 neither stdin nor stderr are a tty device and you have neither a controlling tty nor systemd - can't ask for 'Enter Private Key Password:'. If you used --daemon, you need to use --askpass to make passphrase-protected keys work, and you can not use --auth-nocache.
```
as you are using keys you need to sort the Password for them as the log says

**Reply by locutus2000** [reply]
Then can you help me into the right direction? I have the username and password set in the credentials file that I mount in the yml file. The OVPN is updated to point to that file and still I get this error. I can connect to the openvpn perfectly fine from a different system.

**Reply by pkishino** [reply]
The username and password are for ovpn.. according to the log you are using some private key that needs a password, this is different.. perhaps if you can share your .ovpn config I can see something

**Reply by locutus2000** [reply]
I have tried askpass and user-pass (if I remember correctly)

`client
dev tun
proto udp
remote xxx 1194
nobind
remote-cert-tls server
tls-version-min 1.2
verify-x509-name xxx name
cipher AES-256-CBC
auth SHA256
askpass
<ca>
-----BEGIN CERTIFICATE-----
x
-----END CERTIFICATE-----
</ca>
<cert>
-----BEGIN CERTIFICATE-----
x
-----END CERTIFICATE-----
</cert>
<key>
-----BEGIN ENCRYPTED PRIVATE KEY-----
x
-----END ENCRYPTED PRIVATE KEY-----
</key>
<tls-crypt>
#
# 2048 bit OpenVPN static key
#
-----BEGIN OpenVPN Static key V1-----
x
-----END OpenVPN Static key V1-----
</tls-crypt>
inactive 3600
ping 10
ping-exit 60
resolv-retry 15
verb 3
remap-usr1 SIGTERM
inactive 3600
ping 10
ping-exit 60
resolv-retry 15
verb 3
remap-usr1 SIGTERM
`

**Comment by pkishino** [comment]
Hmm, haven’t used like that before so would need to look it up tomorrow perhaps.
